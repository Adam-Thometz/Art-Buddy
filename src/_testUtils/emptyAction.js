const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION