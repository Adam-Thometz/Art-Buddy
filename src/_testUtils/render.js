import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
