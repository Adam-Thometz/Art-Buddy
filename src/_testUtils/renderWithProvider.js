import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;