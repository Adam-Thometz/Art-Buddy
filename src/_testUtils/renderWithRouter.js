import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;