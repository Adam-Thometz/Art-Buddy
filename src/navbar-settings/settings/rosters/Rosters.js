import React from 'react';
import useRoster from '_hooks/useRoster';

import { useSelector, useDispatch } from 'react-redux';
import { setRoster } from '_redux/settings/mainSettingsActions';

import './Rosters.css';

import NewRoster from './new-roster/NewRoster';
import AddIcon from '_components/icon/add-icon/AddIcon';
import Popup from '_components/popup/Popup';
import Dropdown from '_components/dropdown/Dropdown';

const Rosters = () => {
  const { roster } = useSelector(state => state.mainSettings);
  const [rosters] = useRoster();
  const dispatch = useDispatch();

  const handleSetRoster = e => {
    const roster = rosters[e.target.id];
    dispatch(setRoster(roster));
  };

  const createRosterSelection = () => {
    const rosterSelection = {};
    for (let rosterId in rosters) {
      rosterSelection[rosterId] = rosters[rosterId].name;
    };
    return rosterSelection;
  };

  return (
    <div className='Rosters'>
      <Popup
          title='NEW ROSTER'
          trigger={<AddIcon text='ADD' size='42px' />}
          triggerClass='Settings-add-roster'
          popup={<NewRoster />}
        />
        {Object.keys(rosters).length ? (
          <Dropdown labelText={roster.name ? roster.name : 'SELECT ROSTER'} onClick={handleSetRoster} options={createRosterSelection()} />
        ) : null}
    </div>
  );
};

export default Rosters;