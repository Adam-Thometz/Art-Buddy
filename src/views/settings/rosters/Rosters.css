.Rosters {
  display: flex;
  align-items: center;
}
