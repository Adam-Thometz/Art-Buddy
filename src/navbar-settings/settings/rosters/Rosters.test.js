import Rosters from './Rosters';

import { render } from "_testUtils/render";

describe('Rosters component', () => {
  window.localStorage.setItem('rosters', '{}');
  it('renders without crashing', () => {
    render(<Rosters />);
  });

  it('matches the snapshot of Rosters', () => {
    const { asFragment } = render(<Rosters />);
    expect(asFragment()).toMatchSnapshot();
  });
});