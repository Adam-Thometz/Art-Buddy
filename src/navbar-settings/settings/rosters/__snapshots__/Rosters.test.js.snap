// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rosters component matches the snapshot of Rosters 1`] = `
<DocumentFragment>
  <div
    class="Rosters"
  >
    <figure
      class="Icon"
    >
      <img
        alt=""
        src="file"
        style="height: 42px; width: 42px; cursor: pointer;"
      />
      <figcaption
        class="Icon-text"
        style="font-size: 1rem;"
      >
        ADD
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;
